dao-voting/tests/harness.rs

```diff
@@ -0,0 +1,897 @@
+            let call_params = CallParameters::new(
+                Some(asset_amount),
+                Some(AssetId::from(*gov_token_id)),
+                Some(100_000),
+            );
+            user.dao_voting
+                .deposit()
+                .tx_params(tx_params)
+                .call_params(call_params)
+                .call()
+                .await
+                .unwrap()
+                .value;
+
+            assert_eq!(
+                deployer.dao_voting.balance().call().await.unwrap().value,
+                asset_amount
+            );
+
+            assert_eq!(
+                user.dao_voting
+                    .user_balance(Identity::Address(user.wallet.address()))
+                    .call()
+                    .await
+                    .unwrap()
+                    .value,
+                asset_amount
+            );
+
+            user.dao_voting
+                .withdraw(asset_amount * 100)
+                .append_variable_outputs(1)
+                .call()
+                .await
+                .unwrap()
+                .value;
+        }
+    }
+}
+
+mod convert_votes {
+    use super::*;
+
+    mod success {
+        use super::*;
+
+        #[tokio::test]
+        async fn user_can_unlock_tokens() {
+            let (_gov_token, gov_token_id, deployer, user, asset_amount) = setup().await;
+            deployer
+                .dao_voting
+                .constructor(gov_token_id)
+                .call()
+                .await
+                .unwrap();
+
+            assert!(
+                deployer
+                    .gov_token
+                    .unwrap()
+                    .mint_and_send_to_address(100, user.wallet.address())
+                    .append_variable_outputs(1)
+                    .call()
+                    .await
+                    .unwrap()
+                    .value
+            );
+
+            let tx_params = TxParameters::new(None, Some(1_000_000), None, None);
+            let call_params = CallParameters::new(
+                Some(asset_amount),
+                Some(AssetId::from(*gov_token_id)),
+                Some(100_000),
+            );
+            user.dao_voting
+                .deposit()
+                .tx_params(tx_params)
+                .call_params(call_params)
+                .call()
+                .await
+                .unwrap()
+                .value;
+
+            let call_data = get_call_data(gov_token_id);
+
+            user.dao_voting
+                .create_proposal(1, 1, call_data.clone())
+                .call()
+                .await
+                .unwrap()
+                .value;
+
+            user.dao_voting
+                .vote(true, 0, asset_amount / 2)
+                .call()
+                .await
+                .unwrap()
+                .value;
+
+            let proposal = user.dao_voting.proposal(0).call().await.unwrap().value;
+
+            assert_eq!(
```

Comment:
not needed, you should check the user votes instead (before and after to see that it has been unlocked)